service-statistics-unsd/src/main/java/fi/nls/oskari/control/statistics/plugins/unsd/parser/UnsdParser.java

```diff
@@ -0,0 +1,156 @@
+package fi.nls.oskari.control.statistics.plugins.unsd.parser;
+
+import fi.nls.oskari.control.statistics.data.*;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.service.ServiceRuntimeException;
+import fi.nls.oskari.util.JSONHelper;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.*;
+
+public class UnsdParser {
+
+    private final static Logger LOG = LogFactory.getLogger(UnsdParser.class);
+
+    private static final String LANG = "en";
+
+    public List<StatisticalIndicator> parseIndicators(String indicatorsResponse) {
+        try {
+            JSONArray indicators = new JSONArray(indicatorsResponse);
+            return parseIndicators(indicators);
+        } catch (JSONException e) {
+            LOG.error("Error parsing UNSD indicators: " + e.getMessage(), e);
+        }
+        return Collections.emptyList();
+    }
+
+    public List<StatisticalIndicator> parseIndicators(JSONArray goals) {
+        // array of goal.targets[target.indicators]
+        List<StatisticalIndicator> indicatorList = new ArrayList<>();
+        if (goals == null) {
+            return indicatorList;
+        }
+        try {
+            for (int i = 0; i < goals.length(); i++) {
+                JSONObject item = goals.getJSONObject(i);
+                parseIndicatorsResultItem(item)
+                        .ifPresent(list -> indicatorList.addAll(list));
+            }
+        } catch (JSONException e) {
+            LOG.error("Error parsing UNSD indicators: " + e.getMessage(), e);
```

Comment:
We break the loop by catching the exception, but if we've successfully parsed some `StatisticalIndicator`s before the exception we still return the list as if the response was OK.